pkg/client/restclient/plugin.go

```diff
@@ -30,9 +30,16 @@ type AuthProvider interface {
 	// WrapTransport allows the plugin to create a modified RoundTripper that
 	// attaches authorization headers (or other info) to requests.
 	WrapTransport(http.RoundTripper) http.RoundTripper
+	// Login is expected to maybe require user interaction; it won't be called
+	// in environments where a user is not present
+	Login() error
```

Comment:
> @deads2k: Ping on this question. Do we need to disambiguate the flows at the plugin level? Can't we just defer to letting the plugin modify its config however it likes (either interactively or non-interactively)?

I feel like I'm missing something.  If this requires user interaction, they'll need an `io.Reader`, right?  I could live with a `nil` reader meaning, "non-interactive", but they'll still need one, right?